File: c/euler1_3.c
// Euler1 in C
#include <stdio.h>
#include <math.h>

double mySum(int n, int size) {
	return (pow(floor(size/n), 2) + floor(size/n)) * n / 2;
}

int euler1(int size) {
    return mySum(3, size) + mySum(5, size) - mySum(15, size);
}

int main() {
    printf("euler1 = %i\n", euler1(999));
	return 0;
}

File: haskell/euler1_3.hs
-- Euler1 in Haskell

mySum [] = 0
mySum (x:xs) = x + mySum xs

euler1 (n) = mySum [x | x <- [0..n], x `mod` 3 == 0 || x `mod` 5 == 0]

main = do
    let result = euler1 999 
    putStrLn $ "euler1 = " ++ show result

